### PR TITLE
feat: Added nickname logs

### LINKED ISSUE #496: Log usernames/nicknames changes
`s!names @user`

Shows the past nicknames and usernames of the person.

Example code I use in ordo: https://hastebin.com/akexamokon.js https://hastebin.com/woruyicayo.js

### PR DESCRIPTION
**Pull request by kyranet**
Fixes #496

### CODE REVIEW

File: src/events/rawGuildMemberUpdate.ts [hunk 1]
```diff
@@ -1,26 +1,48 @@
...
+
+		// Get the previous nickname
+		guild.nicknames.set(data.user.id, next);
+
+		// If the previous was unset or it's the same as the next one, skip
+		if (typeof previous === 'undefined' || previous === next) return;
 
+		if (guild.settings.get(GuildSettings.Events.MemberNicknameUpdate)) {
+			this.client.emit(Events.GuildMessageLog, MessageLogsEnum.Member, guild, () => new MessageEmbed()
+				.setColor(0xDCE775)
```
Comment (vladfrangu) [inline src/events/rawGuildMemberUpdate.ts:37]: Should probably be in an enum of sorts

File: src/events/rawUserUpdate.ts [hunk 2]
```diff
@@ -12,6 +17,28 @@ export default class extends Event {
...
+		const previous = this.client.userTags.get(data.id);
+		const next = this.client.userTags.create(data);
+
+		if (typeof previous === 'undefined' || previous.username === next.username) return;
+
+		for (const guild of this.client.guilds.values()) {
+			if (!guild.nicknames.has(data.id)) continue;
+			if (guild.settings.get(GuildSettings.Events.MemberNicknameUpdate)) {
+				this.client.emit(Events.GuildMessageLog, MessageLogsEnum.Member, guild, () => new MessageEmbed()
+					.setColor(0xDCE775)
```
Comment (vladfrangu) [inline src/events/rawUserUpdate.ts:34]: Ditto

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/events/rawGuildMemberUpdate.ts (hunk 1 above)
Comment (kyranet) [reply src/events/rawGuildMemberUpdate.ts:37]: Related to #491, it's pending to do and will be done in another Pull Request.

File: src/events/rawUserUpdate.ts (hunk 2 above)
Comment (kyranet) [reply src/events/rawUserUpdate.ts:34]: Ditto

**Review by PyroTechniac — CHANGES_REQUESTED** [review]
Settings menu keys need added

**Comment by kyranet** [comment]
The language keys will be added later alongside a few other.

**Pull request merged**